src/main/java/reposense/parser/CsvParser.java

```diff
@@ -0,0 +1,126 @@
+package reposense.parser;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import reposense.dataobject.Author;
+import reposense.dataobject.RepoConfiguration;
+import reposense.frontend.CliArguments;
+import reposense.util.Constants;
+
+/**
+ * Parses a CSV configuration file for repository information.
+ */
+public class CsvParser {
+    private static final String MESSAGE_UNABLE_TO_READ_CSV_FILE = "Unable to read the supplied CSV file.";
+    private static final String MESSAGE_MALFORMED_LINE_FORMAT = "Warning! line %d in configuration file is malformed.\n"
+            + "Contents: %s";
+
+    private static final int SKIP_FIRST_LINE = 1;
+
+    /** Positions of the elements of a line in the user-supplied CSV file */
+    private static final int ORGANIZATION_POSITION = 0;
+    private static final int REPOSITORY_NAME_POSITION = 1;
+    private static final int BRANCH_POSITION = 2;
+    private static final int GITHUB_ID_POSITION = 3;
+    private static final int DISPLAY_NAME_POSITION = 4;
+    private static final int ALIAS_POSITION = 5;
+
+    /**
+     * Returns a list of {@code RepoConfiguration}, which are the inflated object a line of the csv file.
+     * @throws IOException if user-supplied csv file does not exists or is not readable.
+     */
+    public static List<RepoConfiguration> parse(CliArguments argument) throws IOException {
+        assert (argument != null);
+
+        final Date sinceDate = argument.getSinceDate().orElse(null);
+        final Date untilDate = argument.getUntilDate().orElse(null);
+        final Path path = argument.getConfigFilePath();
+
+        List<RepoConfiguration> repoConfigurations = new ArrayList<RepoConfiguration>();
+        int lineNumber = 1;
+
+        try {
+            // Skips first line, which is the header row
+            final Collection<String> lines = Files.lines(path).skip(SKIP_FIRST_LINE).collect(Collectors.toList());
+
+            for (final String line : lines) {
+                processLine(repoConfigurations, sinceDate, untilDate, line, lineNumber);
+                lineNumber++;
+            }
+
+        } catch (IOException ioe) {
+            throw new IOException(MESSAGE_UNABLE_TO_READ_CSV_FILE);
+        }
```

Comment:
See if you can preserve the original exception as well. Otherwise we could be losing important information useful for debugging.